### DISCUSSION TITLE
Fields in Data Model page should show information such as datatype, not-null, unique and unique-together

### DISCUSSION THREAD

**Discussion by ravi-2912** [directus/directus#13674] [Feature Requests]
Just like required fields are indicated via aserix * and interface is shown when hovered over the field, the other attributes of the field should also be indicated though a text or visual description. An image is shown below. This would then help us get quick view of what the field is constructed of instead of opening up the field sidebar every time.

The following information is desirable
1. data type for e.g. string with length (string-255, string-53), text, json, etc.
2. unique constraint and unique together constraint
3. if dropdown then what number of option to select e.g. (select-5, select-3). 

The default `status` field for e.g. can be given given a text `string-255 select-3 not-null`.

An image of what I am looking for is shown below for reference.

![image](https://user-images.githubusercontent.com/25941167/171511079-76b74ca6-51d7-4ab0-b8cf-eb6b5a170b35.png)

**Comment by benhaynes** [comment]
I like it. I could imagine these attributes using a CSV of "tags" styling to show their config.

One thought I've had for a while is decoupling the **Schema** view from the **Form Layout** view. This would allow us to have a table of database columns for configuring, with full-width real-estate to show this new info... and a dedicated Form Layout view for the GUI of it all. I'd love your thoughts on the mockups I made below:
![Settings-Data-Model-Collection](https://user-images.githubusercontent.com/522079/171513506-28a7115b-5629-4fcb-8c76-da78329bf8a1.png)
![Settings-Data-Model-Schema](https://user-images.githubusercontent.com/522079/171513515-87d4b8eb-7ad3-4f1e-8e89-f3ed27ade0e2.png)

**Reply by ravi-2912** [reply]
I absolutely love the **Form Layout** and **Schema** views. That is miles ahead. Kudos on that.

One thing I noted is that new fields can be added progressively in last row. I particularly like the how I configure fields in current version which is through a sidebar/sidepane. Maybe I just learned enough to use it and am getting used to it.

But if we don't miss on the current functionality and the level of detail we get in customising the fields in current version, this Schema layout could work. I have noted that initially I would just model the fields using bare minimum attributes like type, nullable, unique, etc.  but thesw alone are not enough for me. 

Directus goes beyond the no-code approach wherer bare minimum attributes are actually and significantly complemented by selecting their interface, displays, views, conditions, etc. This is what I like to see in the new **Schema** view if it can happen but also not missing out bare minimum attributes when needed.

On that note, I would also like to see how this layout could work if we have **unique together** or **multi-columns unique** constraints.

**Reply by benhaynes** [reply]
GReat to hear! Yes, these mockups are very "divergent" and exploratory... we would 100% want to keep the drawer for field setup, especially for the advanced setup. This is me mostly thinking through other approaches we could look into. This one specifically tries to match what is handled by a "Database Admin Tool" client... but that doesn't take into consideration all the special sauce that Directus layers on top of the pure SQL schema.

Indices, compound keys, and multi-column constraints should be considered in the design... but are being officially tracked in other Discussions. 🙂

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
